src/pages/index.js

```diff
@@ -68,51 +98,138 @@ cardPreviewPopup.close();
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.updateLikeCount(card.likes);
+
+                    // Deactivate like icon
+                    cardElement.removeLikeIcon();
+                })
+            } else {
+
+                // Add like to the server if user has not liked the card
+                api.addLike(data._id)
+                .then((card) => {
+
+                    // Update like count
+                    cardElement.updateLikeCount(card.likes);
+
+                    // Activate like icon
+                    cardElement.addLikeIcon();
+                })
+            }
+        }
+    }, 
+    selectors.cardTemplate, 
+    userId);
+
+    // Display the card
+    return cardElement.getView();
+}
 
-// Render the initial list of cards on the page
-cardSection.renderItems(initialCards);
+let cardSection;
+let userId;
+
+// Get the app's information and make sure promises are loaded in the correct sequence
+api.getAppInfo()
+    .then(([cards, userData]) => {
+
+        // Find the user id
+        userId = userData._id;
+        userName.textContent = userData.name;
+        userProfession.textContent = userData.about;
+        userImage.src = userData.avatar;
+
+        // Create cards section
+        cardSection = new Section(
+            {
+                items: cards,
+                renderer: (card) => {
+
+                    // Create a new card
+                    const cardElement = createCard(card, userId);
+        
+                    // Display each card
+                    cardSection.addItem(cardElement);
+                },
+            },
+            selectors.cardsList,
+            userId
+        );
+
+        // Render the entire list of cards on the page
+        cardSection.renderItems(cards);
+    })
 
 /* -------------------------------------------------------------------------- */
 /*                                  Add Form                                  */
 /* -------------------------------------------------------------------------- */
 
-
 // Create the add form instance
 const addFormPopup = new PopupWithForm(selectors.addFormPopup, (formData) => {
-
-    // Create a new card
-    const newCard = createCard(formData);
     
-    // Close the add form
-    addFormPopup.close();
+    // Render loading status
+    renderLoading(true, selectors.addFormButton);
+
+    // Add the new card to the server
+    api.addNewCard(formData)
+    .then((formData) => {
+
+        // Create a new card
+        const newCard = createCard(formData, userId);
+
+        // Add the new card to the section
+        cardSection.addItem(newCard);
+    })
+    .finally(() => {
+
+        // Restore pre-loading status
+        renderLoading(false, selectors.addFormButton);
 
-    // Add the new card to the section
-    cardSection.addItem(newCard);
+        // Close the add form
+        addFormPopup.close();
```

Comment:
Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Fix that for every request

Reply:
Ah, of course! That's fixed for every request. Thank you. 🤓